src/Orleans.Runtime/Storage/StateStorageBridge.cs

```diff
@@ -19,8 +19,17 @@ public class StateStorageBridge<TState> : IStorage<TState>
 
         public TState State
         {
-            get { return grainState.State; }
-            set { grainState.State = value; }
+            get
+            {
+                CheckRuntimeContext();
+                return grainState.State;
+            }
+
+            set
+            {
+                CheckRuntimeContext();
+                grainState.State = value;
+            }
```

Comment:
These are guard rails to help ensure the user doesn't violate scheduling rules in an activation. We do similarly for the `IGrainRuntime` impl.